server/routes/utils/dataReportHelpers.ts

```diff
@@ -309,22 +309,28 @@ function traverse(data: any, keys: string[], result: { [key: string]: any } = {}
       const flattenedValues: { [key: string]: any[] } = {};
 
       Object.keys(data).forEach((dataKey) => {
-        if (dataKey.startsWith(key + '.')) {
+        const normalizedDataKey = dataKey.replace(/^_source\./, '').replace(/\[\d+\]/g, '');
+        const normalizedKey = key.replace(/\[\d+\]/g, '');
+
+        if (normalizedDataKey === normalizedKey || normalizedDataKey.startsWith(normalizedKey + '.')) {
           result[dataKey] = data[dataKey];
         }
+
         const arrayValue = data[dataKey];
         if (Array.isArray(arrayValue)) {
-          arrayValue.forEach((item) => {
+          arrayValue.forEach((item, index) => {
             if (typeof item === 'object' && item !== null) {
-              Object.keys(item).forEach((subKey) => {
-                const newKey = `${dataKey}.${subKey}`;
-                if (!flattenedValues[newKey]) {
-                  flattenedValues[newKey] = [];
-                }
-                flattenedValues[newKey].push(item[subKey]);
-              });
+              flattenNestedObject(item, `${dataKey}[${index}]`, flattenedValues);
+            } else {
+              const newKey = `${dataKey}[${index}]`;
+              if (!flattenedValues[newKey]) {
+                flattenedValues[newKey] = [];
+              }
+              flattenedValues[newKey].push(item);
             }
           });
+        } else if (typeof arrayValue === 'object' && arrayValue !== null) {
+          flattenNestedObject(arrayValue, dataKey, flattenedValues);
```

Comment:
not a blocker but can this be done by fixing `flattenHits`? seems we are doing recursive traversal again on flattened data, meaning flattenHits didn't flatten everything

if flattenHits can correctly flatten arrays, then i think it would reduce code complexity